src/overlays/actors/ovl_En_Famos/z_en_famos.c

```diff
@@ -28,133 +62,811 @@ const ActorInit En_Famos_InitVars = {
+    }
+
+    this->stateTimer--;
+    if (this->stateTimer == 0) {
+        this->actor.world.pos.y = this->baseHeight;
+        EnFamos_SetupChase(this);
+    } else {
+        this->actor.world.pos.y = Math_SinS(this->stateTimer * 0x1000) * 30.0f + this->baseHeight;
+    }
+}
+
+/**
+ * Summary: Famos has spotted the player; Begin chasing to attack.
+ */
+void EnFamos_SetupChase(EnFamos* this) {
+    this->hoverTimer = 0;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->actionFunc = EnFamos_Chase;
+}
+
+void EnFamos_Chase(EnFamos* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    Vec3f abovePlayerPos;
+    u32 surfaceType;
+
+    EnFamos_UpdateBobbingHeight(this);
+    Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0x800);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    abovePlayerPos.x = player->actor.world.pos.x;
+    abovePlayerPos.y = player->actor.world.pos.y + 100.0f;
+    abovePlayerPos.z = player->actor.world.pos.z;
+    this->actor.world.rot.x = -Actor_PitchToPoint(&this->actor, &abovePlayerPos);
+    Math_StepToF(&this->actor.speedXZ, 6.0f, 0.5f);
+
+    surfaceType = func_800C9B18(&globalCtx->colCtx, this->actor.floorPoly, this->actor.floorBgId);
+    if (this->actor.xzDistToPlayer < 30.0f && this->actor.floorHeight > BGCHECK_Y_MIN && // close enough
+        (surfaceType != 0xC && surfaceType != 0xD)) {
+        EnFamos_SetupAttackAim(this);
+
+    } else if (Player_GetMask(globalCtx) == PLAYER_MASK_STONE ||
+               (this->aggroDistance < Actor_XZDistanceToPoint(&GET_PLAYER(globalCtx)->actor, &this->calmPos)) ||
+               !Actor_IsFacingPlayer(&this->actor, 0x6000)) {
+        EnFamos_SetupScanForPlayer(this);
+    }
+}
+
+void EnFamos_SetupAttackAim(EnFamos* this) {
+    Animation_PlayOnce(&this->skelAnime, &gFamosShakeAnim);
+    this->actor.speedXZ = 0.0f;
+    Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_AMOS_VOICE);
+    this->actionFunc = EnFamos_AttackAim;
+}
+
+void EnFamos_AttackAim(EnFamos* this, GlobalContext* globalCtx) {
+    func_800B9010(&this->actor, NA_SE_EN_LAST1_FALL_OLD - SFX_FLAG);
+    if (SkelAnime_Update(&this->skelAnime)) {
+        EnFamos_SetupAttack(this);
+    }
+}
+
+void EnFamos_SetupAttack(EnFamos* this) {
+    this->actor.world.rot.x = -0x4000;
+    this->collider1.base.atFlags |= AT_ON;
+    this->stateTimer = 4;
+    this->actionFunc = EnFamos_Attack;
+}
+
+void EnFamos_Attack(EnFamos* this, GlobalContext* globalCtx) {
+    s32 hitFloor;
+    u32 surfaceType;
+
+    Math_StepToF(&this->actor.speedXZ, 20.0f, 2.0f);
+    this->stateTimer--;
+    if (this->stateTimer == 0) {
+        this->emblemCollider.base.acFlags &= ~AC_ON;
+    }
+
+    surfaceType = func_800C9B18(&globalCtx->colCtx, this->actor.floorPoly, this->actor.floorBgId);
+    hitFloor = this->actor.bgCheckFlags & 1;
+    if (hitFloor || this->actor.floorHeight == BGCHECK_Y_MIN || (surfaceType == 0xC || surfaceType == 0xD)) {
```

Comment:
```suggestion
    if (hitFloor || (this->actor.floorHeight == BGCHECK_Y_MIN) || (surfaceType == 0xC) || (surfaceType == 0xD)) {
```
Unless you think it's semantically valuable to have () round both; it should be equivalent?

Reply:
I do like to visually isolate when one or the other condition are relevant to each other (in this case which surface type would appear to be its own meta condition _if surface type is valid exception_), even if it matches without, but I could remove them all or add to each here, either or seems fine.